### DISCUSSION TITLE
Unable to connect to MQTT broker

### DISCUSSION THREAD

**Discussion by Pocky0013** [tsightler/ring-mqtt#422] [Support Q&A] [answered]
Hi everyone,
I'm using HA OS installed on a Raspberry PI. I managed to generate the RING token, but I can't connect my MQTT server.
I use Mosquitto broker installed on the same machine. the server URL is therefore local and I use the url: mqtt://core-mosquitto:1883 for other services (Zigbee2MQTT).
My final url is therefore: mqtt://ring_mqtt:MyPassword@mqtt://core-mosquitto:1883. But the log gives me the following error: Unable to connect to MQTT broker getaddrinfo ENOTFOUND ring_mqtt. I tried with local IP and url homeassistant.local but nothing works better. Do you know how to help me? Thanks in advance

**Comment by tsightler** [comment, marked as answer]
> mqtt://ring_mqtt:MyPassword@mqtt://core-mosquitto:1883

That's not a valid URL syntax, you don't need to restart the URL after the @ symbol, just continue wit the host/port.  Most likely it should be:

mqtt://ring_mqtt:MyPassword@core-mosquitto:1883

**Reply by Pocky0013** [reply]
Many Thank's all runing good now. 
It's good to know that the password can't have character like : % or #
I change the password whit only letter and number --> login ok !

**Reply by tsightler** [reply]
It's a URL, such characters are not valid in a URL, but you can use URL encoding if you want to have those characters.
